doc/src/manual/memory-management.md

```diff
@@ -0,0 +1,203 @@
+* **Sweep threads**: Used for concurrent sweeping of freed memory (default: 0, disabled)
+
+You can configure GC threading using:
+
+```bash
+julia --gcthreads=4,1  # 4 mark threads, 1 sweep thread
+julia --gcthreads=8    # 8 mark threads, 0 sweep threads
+```
+
+Or via environment variable:
+
+```bash
+export JULIA_NUM_GC_THREADS=4,1
+julia
+```
+
+### Recommendations
+
+For compute-intensive workloads:
+
+* Use multiple mark threads (the default configuration is usually appropriate)
+* Consider enabling concurrent sweeping with 1 sweep thread for allocation-heavy workloads
+
+For memory-intensive workloads:
+
+* Enable concurrent sweeping to reduce GC pauses
+* Monitor GC time using `@time` and adjust thread counts accordingly
+
+## [Monitoring and Debugging](@id man-gc-monitoring)
+
+### Basic Memory Monitoring
+
+Use the `@time` macro to see memory allocation and GC overhead:
+
+```julia
+julia> @time some_computation()
+  2.123456 seconds (1.50 M allocations: 58.725 MiB, 17.17% gc time)
+```
+
+This shows:
+
+* Total execution time
+* Number of allocations
+* Memory allocated
+* Percentage of time spent in garbage collection
+
+### GC Logging
+
+Enable detailed GC logging to understand collection patterns:
+
+```julia
+julia> GC.enable_logging(true)
+julia> # Run your code
+julia> GC.enable_logging(false)
+```
+
+This logs each garbage collection event with timing and memory statistics.
+
+### Manual GC Control
+
+While generally not recommended, you can manually trigger garbage collection:
+
+```julia
+GC.gc()          # Force a garbage collection
+GC.enable(false) # Disable automatic GC (use with caution!)
+GC.enable(true)  # Re-enable automatic GC
+```
+
+**Warning**: Disabling GC can lead to memory exhaustion. Only use this for specific performance measurements or debugging.
+
+## [Performance Considerations](@id man-gc-performance)
+
+### Reducing Allocations
+
+The best way to minimize GC impact is to reduce unnecessary allocations:
+
+* Use in-place operations when possible (e.g., `x .+= y` instead of `x = x + y`)
+* Pre-allocate arrays and reuse them
+* Avoid creating temporary objects in tight loops
+* Consider using `StaticArrays.jl` for small, fixed-size arrays
+
+### Memory-Efficient Patterns
+
+* Use `Vector{Float64}` instead of `Vector{Any}` when possible
+* Avoid global variables that change type
+* Use `const` for global constants
+* Consider memory pools for frequently allocated/freed objects
+
+### Profiling Memory Usage
+
+For detailed guidance on profiling memory allocations and identifying performance bottlenecks, see the [Profiling](@ref man-profiling) section.
+
+## [Advanced Configuration](@id man-gc-advanced)
+
+### Environment Variables
+
+Additional GC-related environment variables:
+
+* `JULIA_GC_WAIT_FOR_DEBUGGER`: Pause for debugger attachment during GC (debug builds only)
```

Comment:
This is documented elsewhere as doing something else (it only matters when crashing and julia is debugging)